src/client/cursor.rs

```diff
@@ -58,46 +59,42 @@ impl <'a> Cursor<'a> {
                     v.push_back(doc.clone());
                 }
 
-                Some((v, cid))
+                Ok((v, cid))
             },
-            _ => None
+            _ => Err(Error::CursorMissingError)
         }
     }
 
-    fn get_bson_and_cid_from_command_message(message: Message) -> Option<(VecDeque<bson::Document>, i64, String)> {
-        match Cursor::get_bson_and_cid_from_message(message) {
-            Some((v, _)) => {
-                if v.len() != 1 {
-                    return None;
-                }
-
-                let ref doc = v[0];
+    fn get_bson_and_cid_from_command_message(message: Message) -> Result<(VecDeque<bson::Document>, i64, String)> {
```

Comment:
The function needs to also return the command namespace, so this is correct. I'll rename this function to make it more accurate.